### ISSUE TITLE
Can't build iPhone app on Mac after installing v3.7.0.768

### ISSUE THREAD

**Issue by torbonde** [MonoGame/MonoGame#5898]
I just downloaded the newest development build (v3.7.0.768) for mac, using the link on monogame.net/downloads. I can now no longer build any MonoGame iPhone/iPad app.

When trying to build, I get the following error message in the output window (using Diagnostic log verbosity):
```
Building Solution: FooBarGame (Debug|iPhoneSimulator)

Build started 27-08-2017 18:53:34.
__________________________________________________
Project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" (Build target(s)):

Initial Properties:
__CF_USER_TEXT_ENCODING = 0x1F5:0x0:0x9
Apple_PubSub_Socket_Render = /private/tmp/com.apple.launchd.uXA77SsSWu/Render
BuildingInsideVisualStudio = true
Configuration = Debug
CurrentSolutionConfigurationContents = <SolutionConfiguration>
<ProjectConfiguration Project="{471536E7-9696-45FC-B93D-34773DB20D4C}" AbsolutePath="/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" BuildProjectInSolution="True">Debug|iPhoneSimulator</ProjectConfiguration>
</SolutionConfiguration>
DeviceSpecificBuild = true
DISPLAY = /private/tmp/com.apple.launchd.xGUI2tcDdB/org.macosforge.xquartz:0
DYLD_FALLBACK_LIBRARY_PATH = /Applications/Xamarin Studio.app/Contents/Resources/lib:/Applications/Xamarin Studio.app/Contents/Resources/lib/monodevelop/bin:/Applications/Xcode.app/Contents/Developer/usr/lib:/Library/Frameworks/Mono.framework/Libraries:/lib:/usr/lib:/usr/local/lib
HOME = /Users/torbonde
LANGUAGE = da
LC_NUMERIC = C
LOGNAME = torbonde
MONO_GAC_PREFIX = /Applications/Xamarin Studio.app/Contents/Resources
MONO_GC_PARAMS = major=marksweep-conc
MONODEVELOP_64BIT_SAFE = yes
MONODEVELOP_STUB_LANGUAGE = da
MSBuildBinPath = /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin
MSBuildExtensionsPath = /Library/Frameworks/Mono.framework/External/xbuild
MSBuildExtensionsPath32 = /Library/Frameworks/Mono.framework/External/xbuild
MSBuildExtensionsPath64 = /Library/Frameworks/Mono.framework/External/xbuild
MSBuildFrameworkToolsPath = /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/4.5
MSBuildFrameworkToolsPath32 = /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/4.5
MSBuildProjectDefaultTargets = Build
MSBuildProjectDirectory = /Users/torbonde/Projects/FooBarGame/FooBarGame
MSBuildProjectExtension = .csproj
MSBuildProjectFile = FooBarGame.csproj
MSBuildProjectFullPath = /Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj
MSBuildProjectName = FooBarGame
MSBuildThisFile = FooBarGame.csproj
MSBuildThisFileDirectory = /Users/torbonde/Projects/FooBarGame/FooBarGame/
MSBuildThisFileDirectoryNoRoot = Users/torbonde/Projects/FooBarGame/FooBarGame/
MSBuildThisFileExtension = .csproj
MSBuildThisFileFullPath = /Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj
MSBuildThisFileName = FooBarGame
MSBuildToolsPath = /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin
MSBuildToolsPath32 = /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin
MSBuildToolsRoot = /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0
MSBuildToolsVersion = 14.0
MtouchVerbosity = 4
NuGetAuthoringPath = /Applications/Xamarin Studio.app/Contents/Resources/lib/monodevelop/AddIns/MonoDevelop.Packaging/Targets
OS = Unix
PATH = /Library/Frameworks/Mono.framework/Commands:/Applications/Xamarin Studio.app/Contents/Resources:/Applications/Xamarin Studio.app/Contents/MacOS:/usr/bin:/bin:/usr/sbin:/sbin
PKG_CONFIG_LIBDIR = 
PKG_CONFIG_PATH = /Applications/Xamarin Studio.app/Contents/Resources/lib/pkgconfig:/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/pkgconfig
Platform = iPhoneSimulator
SHELL = /bin/bash
SolutionDir = /Users/torbonde/Projects/FooBarGame/
SolutionFilename = FooBarGame.sln
SolutionName = FooBarGame
SolutionPath = /Users/torbonde/Projects/FooBarGame/FooBarGame.sln
SSH_AUTH_SOCK = /private/tmp/com.apple.launchd.KLIOsEgpAO/Listeners
TargetiOSDevice = <?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>device</key>
	<dict>
		<key>architecture</key>
		<string>i386, x86_64</string>
		<key>model</key>
		<string>iPhone9,1</string>
		<key>os</key>
		<string>iOS</string>
		<key>os-version</key>
		<string>10.3.1</string>
	</dict>
</dict>
</plist>

TMPDIR = /var/folders/_w/cmbzjv6d1d36zx93mp35xvdh0000gn/T/
UseHostCompilerIfAvailable = false
USER = torbonde
XCODE_DEVELOPER_DIR_PATH = /Applications/Xcode.app/Contents/Developer
XPC_FLAGS = 0x0
XPC_SERVICE_NAME = com.xamarin.xamarinstudio.16684

Initial Items:
/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj: Importing project /Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Common.props from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj: Importing project /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.CSharp.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.CSharp.targets: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets 
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Microsoft.Common.props from extension path /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Microsoft.Common.props: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Imports/Microsoft.Common.props/ImportBefore/Microsoft.NuGet.ImportBefore.props from extension path /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Imports/Microsoft.Common.props/ImportBefore/Microsoft.NuGet.ImportBefore.props: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/Microsoft/NuGet/Microsoft.NuGet.props from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Microsoft.Common.targets/ImportAfter/Microsoft.NuGet.ImportAfter.targets from extension path /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Microsoft.Common.targets/ImportAfter/Microsoft.NuGet.ImportAfter.targets: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/Microsoft/NuGet/Microsoft.NuGet.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/Microsoft.Common.targets/ImportAfter/Microsoft.NuGet.ImportAfter.targets: Importing project /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/NuGet.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.CSharp.targets: Importing project /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets: Importing project /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.props from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets: Importing project /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Analysis.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Analysis.targets: Importing project /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Analysis.Compatibility.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj: Importing project /Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Content.Builder.targets from extension path /Library/Frameworks/Mono.framework/External/xbuild
Building target "_CheckForInvalidConfigurationAndPlatform" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets").
Done building target "_CheckForInvalidConfigurationAndPlatform" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets").
Building target "Build" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets").
Building target "BuildContent" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Content.Builder.targets"); "Build" depends on it.
Building target "Prepare" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Content.Builder.targets"); "BuildContent" depends on it.
: error : Error building target Prepare: Object reference not set to an instance of an object
Error building target Prepare: System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.Build.BuildEngine.Project.GetMetadataBatched (System.String itemName, System.String metadataName) [0x00005] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/Project.cs:1389 
  at Microsoft.Build.BuildEngine.MetadataReference.ConvertToString (Microsoft.Build.BuildEngine.Project project, Microsoft.Build.BuildEngine.ExpressionOptions options) [0x00000] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/MetadataReference.cs:74 
  at Microsoft.Build.BuildEngine.ExpressionCollection.ConvertToString (Microsoft.Build.BuildEngine.Project project, Microsoft.Build.BuildEngine.ExpressionOptions options) [0x0003b] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/ExpressionCollection.cs:147 
  at Microsoft.Build.BuildEngine.ExpressionCollection.ConvertToNonArray (Microsoft.Build.BuildEngine.Project project, System.Type type, Microsoft.Build.BuildEngine.ExpressionOptions options) [0x00000] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/ExpressionCollection.cs:98 
  at Microsoft.Build.BuildEngine.ExpressionCollection.ConvertTo (Microsoft.Build.BuildEngine.Project project, System.Type type, Microsoft.Build.BuildEngine.ExpressionOptions options) [0x00048] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/ExpressionCollection.cs:86 
  at Microsoft.Build.BuildEngine.Expression.ConvertTo (Microsoft.Build.BuildEngine.Project project, System.Type type, Microsoft.Build.BuildEngine.ExpressionOptions options) [0x00000] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/Expression.cs:645 
  at Microsoft.Build.BuildEngine.BuildItem.AddMetadata (System.String name, System.String value) [0x00023] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/BuildItem.cs:283 
  at Microsoft.Build.BuildEngine.BuildItem.Evaluate (Microsoft.Build.BuildEngine.Project project, System.Boolean evaluatedTo) [0x00054] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/BuildItem.cs:311 
  at Microsoft.Build.BuildEngine.BuildTaskItem.Execute () [0x00007] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/BuildTaskItem.cs:64 
  at Microsoft.Build.BuildEngine.TaskBatchingImpl.Execute (Microsoft.Build.BuildEngine.IBuildTask buildTask, Microsoft.Build.BuildEngine.TaskExecutionMode taskExecutionMode) [0x0001c] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/TaskBatchingImpl.cs:104 
  at Microsoft.Build.BuildEngine.TaskBatchingImpl.Build (Microsoft.Build.BuildEngine.IBuildTask buildTask, Microsoft.Build.BuildEngine.TaskExecutionMode taskExecutionMode, System.Boolean& executeOnErrors) [0x0001d] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/TaskBatchingImpl.cs:52 
  at Microsoft.Build.BuildEngine.TargetBatchingImpl.RunTargetWithBucket (System.Collections.Generic.Dictionary`2[TKey,TValue] bucket, Microsoft.Build.BuildEngine.Target target, System.Boolean& executeOnErrors) [0x000ab] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/TargetBatchingImpl.cs:140 
  at Microsoft.Build.BuildEngine.TargetBatchingImpl.Run (Microsoft.Build.BuildEngine.Target target, System.Boolean& executeOnErrors) [0x0004b] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/TargetBatchingImpl.cs:85 
  at Microsoft.Build.BuildEngine.TargetBatchingImpl.Build (Microsoft.Build.BuildEngine.Target target, System.Boolean& executeOnErrors) [0x00016] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/TargetBatchingImpl.cs:64 
  at Microsoft.Build.BuildEngine.Target.DoBuild (System.Boolean& executeOnErrors) [0x00015] in /private/tmp/source-mono-d15-3/bockbuild-d15-3/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/Microsoft.Build.Engine/Microsoft.Build.BuildEngine/Target.cs:279 
Done building target "Prepare" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Content.Builder.targets"); "BuildContent" depends on it.
Done building target "BuildContent" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/External/xbuild/MonoGame/v3.0/MonoGame.Content.Builder.targets"); "Build" depends on it.
Done building target "Build" in project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj" ("/Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/xbuild/14.0/bin/Microsoft.Common.targets").
Done building project "/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj".-- FAILED

Target perfomance summary:
     0,154 ms  _CheckForInvalidConfigurationAndPlatform                1 calls
     0,502 ms  Prepare                                                 1 calls

Tasks perfomance summary:

Build FAILED.
Errors:

/Users/torbonde/Projects/FooBarGame/FooBarGame/FooBarGame.csproj (Build) ->

	: error : Error building target Prepare: Object reference not set to an instance of an object

	 0 Warning(s)
	 1 Error(s)

Time Elapsed 00:00:00.0622150

--------------------- Færdig ---------------------

Build: 1 error, 0 warnings
```

To reproduce: In Xamarin Studio, create new MonoGame iPhone/iPad application and click build.

I'm on macOS Sierra running mono version 5.2.0.215. Before installing the new development build, everything worked just fine.

Do you have any idea of what might be the issue? If not, can you tell me how to uninstall?

I don't really know where to start with this, but a little bit of digging brought me to [this issue](https://github.com/Jjagg/MonoGameContentReferenceTest/issues/3). The author of the issue is also the author of [this PR](https://github.com/MonoGame/MonoGame/pull/4997). It seems related, but I don't know...

I'm happy to provide any logs and so on you might require.

**Comment by dellis1972** [comment]
This is related to https://github.com/MonoGame/MonoGame/pull/5897 
xbuild does not support creating Metadata items in an ItemGroup, we should have used CreateItem instead. That said I think there are still problems with xbuild even with that PR. 

You should be able to just download the 3.6 release and it will overwrite what you have installed.

**Comment by torbonde** [comment]
Alright, thanks for your help!

**Comment by Jjagg** [comment]
@dellis1972 Did this get fixed in #5897? I assume so since we haven't gotten more reports of this (and there must be some people using MG on iOS).

**Comment by harry-cpp** [comment]
This is no longer a valid issue for 3.8.
